catalog/app/containers/Search/Hit.tsx

```diff
@@ -0,0 +1,535 @@
+    marginTop: t.spacing(1),
+  },
+}))
+
+function Secondary({ children }: React.PropsWithChildren<{}>) {
+  const classes = useSecondaryStyles()
+  return <div className={classes.secondary}>{children}</div>
+}
+
+const useDividerStyles = M.makeStyles((t) => ({
+  divider: {
+    marginLeft: t.spacing(0.5),
+    marginRight: t.spacing(0.5),
+  },
+}))
+
+function Divider() {
+  const classes = useDividerStyles()
+  return <span className={classes.divider}> • </span>
+}
+
+interface PackageProps {
+  hit: SearchUIModel.SearchHitPackage
+  showBucket?: boolean
+}
+
+export function Package({ hit, showBucket = false, ...props }: PackageProps) {
+  const { urls } = NamedRoutes.use()
+
+  // this is actually a string, so we need to parse it
+  const metaJson = React.useMemo(() => {
+    if (!hit.meta) return null
+    try {
+      return JSON.parse(hit.meta as any)
+    } catch {
+      return null
+    }
+  }, [hit.meta])
+
+  return (
+    <Card {...props}>
+      <Section grow>
+        <Link to={urls.bucketPackageTree(hit.bucket, hit.name, hit.hash)}>
+          {showBucket && <Heading secondary>{hit.bucket} / </Heading>}
+          <Heading>{hit.name}</Heading>
+        </Link>
+        <Secondary>
+          {readableBytes(hit.size)}
+          <Divider />
+          Updated <Format.Relative value={hit.modified} />
+          {!!hit.workflow?.id && (
+            <>
+              <Divider /> <span style={{ fontWeight: 500 }}>{hit.workflow.id}</span>
+            </>
+          )}
+        </Secondary>
+      </Section>
+
+      {!!hit.comment && (
+        // XXX: render this in a Secondary instead of a seprarate Section?
+        <Section divider>
+          <M.Typography variant="body2">{hit.comment}</M.Typography>
+        </Section>
+      )}
+
+      {!!metaJson && (
+        <Section divider>
+          <JsonDisplay name="Metadata" value={metaJson} />
+        </Section>
+      )}
+    </Card>
+  )
+}
+
+interface ObjectProps {
+  hit: SearchUIModel.SearchHitObject
+  showBucket?: boolean
+}
+
+export function Object({ hit, showBucket = false, ...props }: ObjectProps) {
+  const { urls } = NamedRoutes.use()
+
+  return (
+    <Card {...props}>
+      <Section grow>
+        <Link to={urls.bucketFile(hit.bucket, hit.key, { version: hit.version })}>
+          {showBucket && <Heading secondary>{hit.bucket} / </Heading>}
+          <Heading>{hit.key}</Heading>
+        </Link>
+        <Secondary>
+          {hit.deleted ? 'Delete Marker' : readableBytes(hit.size)}
+          <Divider />
+          {hit.deleted ? 'Deleted' : 'Updated'}{' '}
+          <M.Tooltip title={hit.modified.toLocaleString()}>
+            <span style={{ position: 'relative' }}>
+              <Format.Relative value={hit.modified} />
+            </span>
+          </M.Tooltip>
+          <Divider />
+          <M.Tooltip title={`VersionID: ${hit.version}`}>
+            <span style={{ position: 'relative' }}>v.{hit.version.slice(0, 4)}</span>
```

Comment:
```suggestion
            <span style={{ position: 'relative' }}>VersionId: {trimCenter(hit.version, 12)}</span>
```

Reply:
I think it should be either completely removed, or added some more info: like "VersionId" instead of "v.", and three dots. You can use `text-overflow: ellipsis` for three dots, or `trimCenter` function (though, it needs to be customized with custom divider without spaces).

Reply:
`@ver...sion` works for me too, but I get used to see a version in quilt+s3 URIs

Reply:
well, it doesn't seem terribly useful, but still, so i've decided to keep while de-emphasizing as much as possible.
hint has the whole version string displayed, in case anyone really needs that